proposals/field-keyword-nullability.md

```diff
@@ -23,17 +23,18 @@ The variable denoted by the `field` keyword in a property's accessors is the *ba
 
 The *backing field* has the same type as the property. However, its nullable annotation may differ from the property. To determine this nullable annotation, we introduce the concept of *null-resiliency*. *Null-resiliency* intuitively means that the property's `get` accessor behaves properly even when the field contains the `default` value for its type.
 
-A *field-backed property* is determined to be *null-resilient* or not by performing a special nullable analysis of its get accessor.
-- For the purposes of this analysis, `field` is temporarily assumed to have *annotated* nullability, e.g. `string?`. This causes `field` to have *maybe-null* or *maybe-default* initial state in the get accessor, depending on its type.
+A *field-backed property* is determined to be *null-resilient* or not by performing a special nullable analysis of its `get` accessor.
+- For the purposes of this analysis, `field` is temporarily assumed to have *annotated* nullability, e.g. `string?`. This causes `field` to have *maybe-null* or *maybe-default* initial state in the `get` accessor, depending on its type.
 - Then, if nullable analysis of the getter yields no nullable warnings, the property is *null-resilient*. Otherwise, it is not *null-resilient*.
 - If the property does not have a get accessor, it is (vacuously) null-resilient.
 - If the get accessor is auto-implemented, the property is not null-resilient.
 
 The nullability of the backing field is determined as follows:
-- If the containing property has ***oblivious*** nullability, then the backing field has ***oblivious*** nullability.
-- If the containing property has ***annotated*** nullability (e.g. `string?` or `T?`) or the `[MaybeNull]` attribute, then the backing field has ***annotated*** nullability.
-- If the containing property has *not-annotated* nullability (e.g. `string` or `T`) or the `[NotNull]` attribute, and the property is ***null-resilient***, then the backing field has ***annotated*** nullability.
-- If the containing property has *not-annotated* nullability (e.g. `string` or `T`) or the `[NotNull]` attribute, and the property is ***not null-resilient***, then the backing field has ***not-annotated*** nullability.
```

Comment:
Removing `[NotNull]` is interesting. So, a `[NotNull] string?` property would have a `string?` backing field and the backing field would not have `[field: NotNull]`, if I'm following.

Reply:
I have to contemplate it all a bit. I think I accidentally conflated those bits with the `[field: ...]` bits.

The thing is, all something like `[NotNull] string? Prop { ... }` means is: when a nullable value goes in, a non-null value will come out. It doesn't say whether getter or setter is responsible for doing the sanitizing, so, strictly, we should do the null-resilience check to determine that.

I am leaning toward putting those bits back. Curious about your thoughts.

```cs
[NotNull]
public string? Prop
{
    get => field; // not null-resilient, so field is non-nullable.
    set => field = value; // value and field are maybe-null when exiting setter.
}

[NotNull]
public string? Prop { get; set; } // no warnings today in accessors or in empty constructor. is that fine?
```

Reply:
That all matches my expectations. I'd slightly rather fix up the auto prop similar to other examples where we said we'd want to start erroring for them in situations where we currently ignore at least `field:` attributes.

Reply:
I put the `or [NotNull]` bits back.